bss/batch_submitter.go

```diff
@@ -0,0 +1,220 @@
+package bss
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/ethereum-optimism/optimistic-specs/bss/db"
+	"github.com/ethereum-optimism/optimistic-specs/bss/sequencer"
+	"github.com/ethereum-optimism/optimistic-specs/l2os"
+	"github.com/ethereum-optimism/optimistic-specs/l2os/rollupclient"
+	"github.com/ethereum-optimism/optimistic-specs/l2os/txmgr"
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rpc"
+	hdwallet "github.com/miguelmota/go-ethereum-hdwallet"
+	"github.com/urfave/cli"
+)
+
+const (
+	// defaultDialTimeout is default duration the service will wait on
+	// startup to make a connection to either the L1 or L2 backends.
+	defaultDialTimeout = 5 * time.Second
+)
+
+// Main is the entrypoint into the L2 Output Submitter. This method returns a
+// closure that executes the service and blocks until the service exits. The use
+// of a closure allows the parameters bound to the top-level main package, e.g.
+// GitVersion, to be captured and used once the function is executed.
+func Main(version string) func(ctx *cli.Context) error {
+	return func(ctx *cli.Context) error {
+		cfg := NewConfig(ctx)
+
+		log.Info("Initializing L2 Output Submitter")
+
+		batchSubmitter, err := NewBatchSubmitter(cfg, version)
+		if err != nil {
+			log.Error("Unable to create L2 Output Submitter", "error", err)
+			return err
+		}
+
+		log.Info("Starting L2 Output Submitter")
+
+		if err := batchSubmitter.Start(); err != nil {
+			log.Error("Unable to start L2 Output Submitter", "error", err)
+			return err
+		}
+		defer batchSubmitter.Stop()
+
+		log.Info("L2 Output Submitter started")
```

Comment:
The logs here still refer to an "output submitter", while this is now a separate batch-submitter binary.